recipe/meta.yaml

```
@@ -1,47 +1,50 @@
 {% set name = "scikit-survival" %}
-{% set version = "0.21.0" %}
-
+{% set version = "0.23.1" %}
 
 package:
   name: {{ name|lower }}
   version: {{ version }}
 
 source:
-  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/scikit-survival-{{ version }}.tar.gz
-  sha256: adc744b65983dced8142ec31465389fc738405d589051251e5447aad9a1e02bc
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/scikit_survival-{{ version }}.tar.gz
+  sha256: 9ee5835e6bd74eab69aac0e71511cc4c033d085bed609e5da771dac053ff7e27
 
 build:
   number: 0
   script: {{ PYTHON }} -m pip install . --no-deps --no-build-isolation -vv
-  skip: true  # [py<38]
+  # we do not have osqp for 3.12
+  skip: true  # [py<39 or py>311 ]
+  ignore_run_exports:
+    - numpy
 
 requirements:
   build:
     - {{ compiler("c") }}
     - {{ compiler("cxx") }}
   host:
-    - cython 0.29.32
-    - numpy {{ numpy }}
-    - scikit-learn 1.2.1
-    - setuptools_scm 7.0.4
+    - cython >=3.0.10
+    - numpy >=2.0.0
+    - scikit-learn >=1.4.0,<1.6
```

```suggestion
    - scikit-learn >=1.4.0,<1.5.0  # [py<=312]
    - scikit-learn >=1.5.0,<1.6.0  # [py>312]
```
https://github.com/sebp/scikit-survival/blob/7c8bdb95fc5df4b24e19342d8021501cba3b0cf9/pyproject.toml#L12-L13

---

Actually, as mentioned here https://peps.python.org/pep-0440/#compatible-release
`~=1.4.0` is same as `>=1.4.0,<2.0.0`
Which is fortunate, cause I wouldn't like to rebuild another `scikit-learn` :)

---

My reading is `>=1.4.0,==1.4.*` which we might infer as `>=1.4.0,<1.4.5`.
I see you're trying a few variants.

---

I think it's more of this case:
```
~= 2.2
>= 2.2, == 2.*
```
The trailing `.0` is transparent.

---

The last example they give includes trailing `.0`s:

```
~= 2.2.0
>= 2.2.0, == 2.2.*
```

🤷

---

Oh that's right. I've been looking at poetry syntax.

---

https://peps.python.org/pep-0440/?utm_source=chatgpt.com#compatible-release

actually, my suggestion still stands.

---

Going with 1.5.2 as agreed on slack.